### PR TITLE
Migrate to node 20

### PR DESCRIPTION
**Pull request by etienneburdet**
- @opendatasoft/visualizations-react@0.21.0
 - @opendatasoft/visualizations@0.21.0

## Summary
The goal for this PR is to upgrade the engines requirements for  node.

### Changes
* Disabled rollup visualizer plugin to avoid migrating the full stack.
*  Removed the check for `fetch`, since we don't plan to support node =< 18 and `fetch` is part of the API since 18 on (and should be stable in 21…).
* Replace @mapbox/geo-viewport with @placemarkio/geo-viewport 
* Remove run-p/npm-run-all in favor of lerna/nx
* Upgrade to ChartJS 4 because of some TS problems

I also had to upgrade lerna and storybook, because old version weren't fully compatible with new node. This makes for a few breaking changes:
* No more `lerna bootstrap, add, link` 👉  npm workspaces handle all of that with npm install. Lerna (Nx) is still use for running tasks in parallel (namely watch) and manage versions.
* Storybook now runs on `vite`. It paves the way for actually using vite everywhere, why not migrating to SvelteKit, moving the storybook back to Svelte etc. But for now this is just a minimal port.

Else I just upgrade all references to nodes >=18 (current LTS) and checked it works with node 20.

### Changelog
All packages are now supposed to run on node 18 or 20 (prefered). Support for node 14 and 16 is dropped.

### CODE REVIEW

**Comment by KevinFabre-ods** [comment]
> So last part of the update: `@mapbox/geo-viewport` was crashing and is pretty much abandonned. I decided to drop the computation in favor of a props. Now, there is another way to compute the max zoom as we did with `@mapbox/geo-viewprot` but with `turf` and it would look something like this:
> 
> ```
> const bboxDiagonal = turf.length(turf.bboxPolygon(bounds));
> const screenDiagonal = Math.sqrt(Math.pow(containerWidth, 2) + Math.pow(containerHeight, 2));
> const screenDiagonalMeters = screenDiagonal * 0.0254; // approx ratio for PPI to meters
> const zoom = Math.log2(screenDiagonalMeters / bboxDiagonal) + 8;
> ```
> 
> My idea though, was to implement that in the Studio (leaving the option to pass it more directly if needed). Another option in the Studio would be to pass "hard" values for goeref for exemple. What do you think @KevinFabre-ods ?
> 
> Other differences in chromatic should be really minor.

Looks this comment doesn't fit with the changes. Currently `@mapbox/geo-viewport` has been replaced with `placemark/geo-viewport`. The maxZoom props defined in `ChoroplethVectorTiles.svelte` doesn't do anything as `MapRender` doesn't have a `maxZoom` prop declaration.

**Comment by etienneburdet** [comment]
> > So last part of the update: `@mapbox/geo-viewport` was crashing and is pretty much abandonned. I decided to drop the computation in favor of a props. Now, there is another way to compute the max zoom as we did with `@mapbox/geo-viewprot` but with `turf` and it would look something like this:
> > ```
> > const bboxDiagonal = turf.length(turf.bboxPolygon(bounds));
> > const screenDiagonal = Math.sqrt(Math.pow(containerWidth, 2) + Math.pow(containerHeight, 2));
> > const screenDiagonalMeters = screenDiagonal * 0.0254; // approx ratio for PPI to meters
> > const zoom = Math.log2(screenDiagonalMeters / bboxDiagonal) + 8;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > My idea though, was to implement that in the Studio (leaving the option to pass it more directly if needed). Another option in the Studio would be to pass "hard" values for goeref for exemple. What do you think @KevinFabre-ods ?
> > Other differences in chromatic should be really minor.
> 
> Looks this comment doesn't fit with the changes. Currently `@mapbox/geo-viewport` has been replaced with `placemark/geo-viewport`. The maxZoom props defined in `ChoroplethVectorTiles.svelte` doesn't do anything as `MapRender` doesn't have a `maxZoom` prop declaration.
> 
> Also it's a bit weird that we don't get any eslint error we use a set a prop that doesn't exist. Maybe we should update` eslint-plugin-svelte3` with https://github.com/sveltejs/eslint-plugin-svelte as it is deprecated

Ha sorry, I thought I had posted a comment about that: I dropped the turf idea, it was missing some key projection stuff. And in the meantime I found a maintained fork, which will do for this update. `maxZoom` should go indeed.

**Review by KevinFabre-ods — COMMENTED** [review]
GG for the migration 👏 

I left some comments to have explanation on specific topics, but nothing blocking

File: packages/visualizations/src/components/Chart/Chart.svelte [hunk 1]
```diff
@@ -246,7 +247,10 @@
                 {/if}
             </div>
         {/if}
-        <figure class="chart legend--{legendPosition}">
+        <figure
+            class="chart legend--{legendPosition}"
+            style="--aspect-ratio: {defaultValue(options.aspectRatio, 4 / 3)}"
```
[the remark this answers is not in the document]
Comment (etienneburdet) [reply packages/visualizations/src/components/Chart/Chart.svelte:252]: `aspectRatio` and `maintainAspectRatio` are for the canvas but the not the container. And now ChartJS relies on listening to the parent container size change, so without that charts don't resize without the window now.
See https://www.chartjs.org/docs/latest/configuration/responsive.html

File: packages/visualizations-react/.storybook/main.js
[the remark this answers is not in the document]
Comment (etienneburdet) [reply packages/visualizations-react/.storybook/main.js:1]: Yes, that's probably better, I didn't know about it. This is one is basically copy paste from storybook doc.

Not sure how much we want to dig into Nx though. Right now it's wrapped by Lerna, but we don't really have a Nx conf. Do we want to dive into that?

File: packages/visualizations/src/components/Chart/Chart.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (KevinFabre-ods) [reply packages/visualizations/src/components/Chart/Chart.svelte:252]: I see, it's a bit weird to use twice the aspect ratio but it doesn't seem we can do differently.

.`maintainAspectRatio` and `responsive` are `true` by default I think we can remove: 
```js
// Chart.svelte
chartOptions.maintainAspectRatio = true;
chartOptions.responsive = true;
```

File: packages/visualizations-react/.storybook/main.js
[the remark this answers is not in the document]
Comment (KevinFabre-ods) [reply packages/visualizations-react/.storybook/main.js:1]: Yes but later (in a future story) ;)

**Review by KevinFabre-ods — CHANGES_REQUESTED** [review]
A bit of clean up to do around the package.json and yml files are we should be good 😄

File: .github/workflows/visualizations.yml [hunk 2]
```diff
@@ -35,8 +35,8 @@ jobs:
       - name: Install
         run: npm ci
 
-      - name: Bootstrap
-        run: npm run bootstrap
+      - name: Build
+        run: npm ci --workspaces --include-workspace-root --install-links
```
[the remark this answers is not in the document]
Comment (etienneburdet) [reply .github/workflows/visualizations.yml:39]: Actually, I don't think we even nedd the install step anymore 🤔 Could be in a `build` script indeed

**Comment by etienneburdet** [comment]
I added an `npm run build-ci` script to install/build for all tasks, tell me what you think about it!

File: packages/visualizations-react/package.json [hunk 3]
```diff
@@ -53,7 +51,7 @@
         "trailingComma": "es5"
     },
     "dependencies": {
-        "@opendatasoft/visualizations": "0.21.1",
+        "@opendatasoft/visualizations": "^0.21.1-beta.0",
```
[the remark this answers is not in the document]
Comment (KevinFabre-ods) [reply packages/visualizations-react/package.json]: Did the update here
https://github.com/opendatasoft/ods-dataviz-sdk/pull/228/commits/f76087de35ffed0e1e768e4da654857653d5fc50

**Review by KevinFabre-ods — APPROVED** [review]
Looks fine, I just added a [commit](https://github.com/opendatasoft/ods-dataviz-sdk/pull/228/commits/f76087de35ffed0e1e768e4da654857653d5fc50) to restore the good packages versions. Since your started this branch, we had some releases

**Pull request merged**